### DISCUSSION TITLE
Upgrade to TCA 1.7.1 and build with xcodebuild failed

### DISCUSSION THREAD

**Discussion by BaochengHan-TomTom** [pointfreeco/swift-composable-architecture#2777] [Q&A]
Hi there,

I have upgraded the TCA from 1.2 to 1.7.1 in my project and everything works with Xcode. But the build stops working on my CI/CD as `xcodebuild` is used there. Tried the command locally and it fails with the same errors.

The command used on CI/CD is 
```shell
xcodebuild \
          -project "${{ env.project }}" \
          -scheme "${{ env.scheme }}" \
          -disableAutomaticPackageResolution \
          clean build-for-testing \
          CODE_SIGN_IDENTITY="Apple Development" CODE_SIGNING_REQUIRED=NO \
          -derivedDataPath Build/ \
          -destination 'platform=iOS Simulator,name=iPhone 15 Pro,OS=17.2'
```
The error messages are 
```
Build/SourcePackages/checkouts/swift-case-paths/Package@swift-5.9.swift: error: Signing for "CasePathsMacros" requires a development team. Select a development team in the Signing & Capabilities editor. (in target 'CasePathsMacros' from project 'swift-case-paths')
Build/SourcePackages/checkouts/swift-composable-architecture/Package@swift-5.9.swift: error: Signing for "ComposableArchitectureMacros" requires a development team. Select a development team in the Signing & Capabilities editor. (in target 'ComposableArchitectureMacros' from project 'swift-composable-architecture')
Build/SourcePackages/checkouts/swift-dependencies/Package@swift-5.9.swift: error: Signing for "DependenciesMacrosPlugin" requires a development team. Select a development team in the Signing & Capabilities editor. (in target 'DependenciesMacrosPlugin' from project 'swift-dependencies')
Build/SourcePackages/checkouts/swift-perception/Package.swift: error: Signing for "PerceptionMacros" requires a development team. Select a development team in the Signing & Capabilities editor. (in target 'PerceptionMacros' from project 'swift-perception') 
```
It seems that it has something to do with signing, any clue why? Thanks in advance.

**Comment by mbrandonw** [comment]
I've personally never seen that before, but the most common problem with macros and CI is when you forget to disable validation. Have you done that? If not, [here](https://pointfreeco.github.io/swift-composable-architecture/main/documentation/composablearchitecture/reducer()#CI-build-failures) are the docs.

**Reply by BaochengHan-TomTom** [reply]
Did try both 
```
defaults write com.apple.dt.Xcode IDESkipMacroFingerprintValidation -bool YES
```
and 
```
xcodebuild -skipMacroValidation
```
and the error remains.

Is there any change made on `CODE_SIGNING_*` for those packages in TCA after 1.2? It is strange because neither Xcode nor xcodebuild was updated from my side and with TCA 1.2 everything works well.

**Reply by mbrandonw** [reply]
SPM packages don't have any notion of code signing. That is all done in Xcode, so from a code signing perspective there is nothing different between 1.2 and 1.7 (or any TCA version really).

What version of Xcode is your CI using?

**Reply by BaochengHan-TomTom** [reply]
It is `Version 15.2 (15C500b)`

**Reply by sssuourabh** [reply]
@mbrandonw and @BaochengHan-TomTom i am also facing the same issue. If you have solved please help me.
`Testing failed:
	Signing for "ComposableArchitectureMacros" requires a development team. Select a development team in the Signing & Capabilities editor.
	Signing for "PerceptionMacros" requires a development team. Select a development team in the Signing & Capabilities editor.
	Signing for "DependenciesMacrosPlugin" requires a development team. Select a development team in the Signing & Capabilities editor.
	Signing for "CasePathsMacros" requires a development team. Select a development team in the Signing & Capabilities editor.
	Undefined symbols:
	Linker command failed with exit code 1 (use -v to see invocation)
	Signing for "PerceptionMacros" requires a development team. Select a development team in the Signing & Capabilities editor.
	Signing for "DependenciesMacrosPlugin" requires a development team. Select a development team in the Signing & Capabilities editor.
	Signing for "CasePathsMacros" requires a development team. Select a development team in the Signing & Capabilities editor.
	Signing for "ComposableArchitectureMacros" requires a development team. Select a development team in the Signing & Capabilities editor.
	Library '/Users/userName/Library/Developer/Xcode/DerivedData/AppName-cgsewwyhdpabglckigcascumgbuf/Build/Products/Debug-iphonesimulator/AppName.app//AppName' not found
	Testing cancelled because the build failed.`

**Reply by stephencelis** [reply]
@sssuourabh I think without a sample project that reproduces the issue it will be difficult for anyone to help. It sounds like a general project configuration issue, though, and not a TCA issue, other than the fact that TCA includes a macro.

**Comment by sssuourabh** [comment]
@stephencelis I haven't done anything but now the build is compiling and tests are passing. Thanks for your response.
